### PR TITLE
Switch to Commonmark

### LINKED ISSUE #279: Migrate to a full-PHP stack
We've discussed in the past a lot about not using Ruby anymore for generating the static site. Up until now, the main objection (pushed forward by @stefanotorresi who migrated it) was that there was no Markdown native PHP parser that supported GitHub-flavored MD, and a code highlighting library that could do the same as Ruby, so a full server-side work, without the need of a JS lib on the browser side.

Today I've found the two libs that we would need to do that!
 * CommonMark (since version 1.3: https://www.colinodell.com/blog/202002/league-commonmark-130-adds-full-gfm-support)
 * https://github.com/spatie/commonmark-highlighter

Yay! This should mean faster local generation times, reduced Docker overhead, faster deploy times!

### PR DESCRIPTION
**Pull request by Jean85**
Fixes #279. 

Still WIP: I've configured Commonmark and it seems to work, but the highlight doesn't; maybe I need to cleanup the SCSS/Ruby stuff, but I need some time to figure it out. Anyone has some suggestions?

### CODE REVIEW

**Comment by stefanotorresi** [comment]
It's been a bit too long, I cannot remember at all :)

**Comment by goetas** [comment]
if common mark is fast enough, then you can remove the https://github.com/php-fig/www.php-fig.org/blob/master/app/config/sculpin_kernel_dev.yml.example i added for faster local development

File: app/config/sculpin_kernel.yml [hunk 1]
```diff
@@ -31,17 +31,32 @@ sculpin:
...
     tags:
       - { name: twig.extension }
+
+  League\CommonMark\GithubFlavoredMarkdownConverter:
+    arguments:
+      - []
+      - '@League\CommonMark\ConfigurableEnvironmentInterface'
+
+  League\CommonMark\ConfigurableEnvironmentInterface:
+    factory: ['\League\CommonMark\Environment', 'createCommonMarkEnvironment']
```
Comment (goetas) [inline app/config/sculpin_kernel.yml:52]: Is the initial  `\`  required?

Comment (Jean85) [reply app/config/sculpin_kernel.yml:52]: Nope! Removing it.

**Comment by Jean85** [comment]
[EDIT] I've revamped & updated with master this PR, and now markdown parsing is fully functioning (fixed in 76efd86). 
Highlight.js CSS is via CDN now, we should bundle it.

Also, we could replace SASS parsing (maybe in separate PR?) and update the PHP container to have latest PHP patch & Composer 2.

**Pull request merged**